miden-tx/tests/integration/bench/tx_benchmark.rs

```diff
@@ -0,0 +1,156 @@
+use crate::MockDataStore;
+use miden_lib::transaction::ToTransactionKernelInputs;
+use miden_objects::{accounts::AccountStub, transaction::TransactionArgs};
+use miden_tx::{TransactionExecutor, TransactionHost};
+use vm_processor::{
+    AdviceExtractor, AdviceInjector, AdviceProvider, ExecutionError, ExecutionOptions, Host,
+    HostResponse, ProcessState, RecAdviceProvider,
+};
+
+// CONSTANTS
+// ================================================================================================
+
+const PROLOGUE_START: u32 = 0x2_0000; // 131072
+const PROLOGUE_END: u32 = 0x2_0001; // 131073
+
+const NOTE_PROCESSING_START: u32 = 0x2_0002; // 131074
+const NOTE_PROCESSING_END: u32 = 0x2_0003; // 131075
+
+const TX_SCRIPT_PROCESSING_START: u32 = 0x2_0004; // 131076
+const TX_SCRIPT_PROCESSING_END: u32 = 0x2_0005; // 131077
+
+const EPILOGUE_START: u32 = 0x2_0006; // 131078
+const EPILOGUE_END: u32 = 0x2_0007; // 131079
+
+const SPAN_CREATION_SHIFT: u32 = 2;
+
+// BENCHMARK HOST
+// ================================================================================================
+
+/// Wrapper around [TransactionHost] used for benchmarking
+pub struct BenchHost<A: AdviceProvider> {
+    host: TransactionHost<A>,
+    prologue_start: Option<u32>,
+    note_processing_start: Option<u32>,
+    tx_script_processing_start: Option<u32>,
+    epilogue_start: Option<u32>,
```

Comment:
Let's define a separate struct for these - maybe something like `TransactionProgress`. We may even move it into the main `TransactionHost` in a follow-up PR.